### PR TITLE
Pm 44 user preference functionality

### PR DESCRIPTION
**Pull request by xHoudek**
- User can save preferences after selecting them in the filter
- Shelters and logged out users cannot save preferences
- When pet page or shelter pet page loads, filters are taken from DB and automatically applied
- User can clear filters and save to effectively remove preferences (more delete options coming in PM-54)

### CODE REVIEW

File: src/components/PetList.js [hunk 1]
```diff
@@ -55,7 +68,54 @@ export default function PetList(props) {
     { value: 'houseTrained', label: 'House Trained' },
   ];
 
+  const savePreferences = async (e) => {
+    e.preventDefault();
+    const objectKeys = ['typeOfAnimal', 'breed', 'sex', 'minAge', 'maxAge', 'more'];
+    const preferenceTypes = [typeOfAnimal, breed, sex.value, minAge, maxAge, more];
```
Comment (dotte-osu) [inline src/components/PetList.js]: I think `sex.value` might cause null exception when there is no sex assigned.

File: src/components/PetList.js [hunk 2]
```diff
@@ -55,7 +68,54 @@ export default function PetList(props) {
...
+          sex: res.Sex ? { value: res.Sex, label: res.Sex } : '',
+          minAge: res.MinAge ? res.MinAge : '',
+          maxAge: res.MaxAge ? res.MaxAge : '',
+          more: res.More ? JSON.parse(res.More) : []
+        }
+          setSavedPrefs(true);
+          savedPreferencesHandler(changedPrefs);
+      }
+    } catch (error) {
+      console.log(error);
```
Comment (dotte-osu) [inline src/components/PetList.js]: debug.log(error)?

File: src/components/PetList.js [hunk 3]
```diff
@@ -55,7 +68,54 @@ export default function PetList(props) {
...
+
+        const changedPrefs = {
+          typeOfAnimal: res.TypeOfAnimal ? JSON.parse(res.TypeOfAnimal) : [],
+          breed: res.Breed ? JSON.parse(res.Breed) : [],
+          sex: res.Sex ? { value: res.Sex, label: res.Sex } : '',
+          minAge: res.MinAge ? res.MinAge : '',
+          maxAge: res.MaxAge ? res.MaxAge : '',
+          more: res.More ? JSON.parse(res.More) : []
+        }
+          setSavedPrefs(true);
```
Comment (dotte-osu) [inline src/components/PetList.js]: indentation is off? maybe?

File: src/components/PetList.js [hunk 4]
```diff
@@ -92,7 +160,10 @@ export default function PetList(props) {
                 id='typeOfAnimal'
                 value={typeOfAnimal}
                 options={typeOptions}
-                onChange={selectedOptions => inputChangeHandler('typeOfAnimal', selectedOptions)}
+                onChange={selectedOptions => {
+                  setConfirmSaved(false);
+                  inputChangeHandler('typeOfAnimal', selectedOptions)
+                }}
                 isMulti
```
Comment (dotte-osu) [inline src/components/PetList.js:169]: this is new! interesting!

**Review by dotte-osu — CHANGES_REQUESTED** [review]
Looks really good! There are a few tiny things to fix but everything looks pretty good. I didn't test about the null exception so I might be wrong.

File: src/components/PetList.js (hunk 1 above)
Comment (xHoudek) [reply src/components/PetList.js]: Good catch!

**Review by dotte-osu — APPROVED** [review]
Approved

**Pull request merged**